File: eclipse-scout-core/src/table/columns/CompactLine.js
/*
 * Copyright (c) 2010-2020 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
import {Cell, CompactLineBlock} from '../../index';

export default class CompactLine {

  constructor(labelOrHeaderCell, textOrCell) {
    /** @type {CompactLineBlock} */
    this.labelBlock = null;
    /** @type {CompactLineBlock} */
    this.textBlock = null;
    if (labelOrHeaderCell instanceof Cell) {
      this.setLabelBlock(this.convertHeaderCellToBlock(labelOrHeaderCell));
    } else {
      this.setLabelBlock(new CompactLineBlock(labelOrHeaderCell));
    }
    if (textOrCell instanceof Cell) {
      this.setTextBlock(this.convertCellToBlock(textOrCell));
    } else {
      this.setTextBlock(new CompactLineBlock(textOrCell));
    }
  }

  convertHeaderCellToBlock(cell) {
    return this.convertCellToBlock(cell);
  }

  convertCellToBlock(cell) {
    let block = new CompactLineBlock();
    if (cell != null) {
      block.setText(cell.text);
      block.setIcon(cell.iconId);
      block.setEncodeHtmlEnabled(!cell.htmlEnabled);
    }
    return block;
  }

  setLabelBlock(block) {
    scout.assertParameter('block', block);
    this.labelBlock = block;
  }

  setTextBlock(block) {
    scout.assertParameter('block', block);
    this.textBlock = block;
  }

  build() {
    let label = this.labelBlock.build();
    if (label) {
      label += ': ';
    }
    let value = this.textBlock.build();
    if (!value) {
      return '';
    }
    return `
<div class="compact-cell-line">
  <span class="compact-cell-line-label">${label}</span>
  <span class="compact-cell-line-value">${value}</span>
</div>`;
  }
}

File: org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/basic/table/columns/CompactLine.java
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.scout.rt.client.ui.basic.table.columns;

import static org.eclipse.scout.rt.platform.html.HTML.*;

import org.eclipse.scout.rt.client.ui.basic.cell.ICell;
import org.eclipse.scout.rt.client.ui.basic.table.IHeaderCell;
import org.eclipse.scout.rt.platform.html.HTML;
import org.eclipse.scout.rt.platform.util.Assertions;
import org.eclipse.scout.rt.platform.util.StringUtility;

public class CompactLine {

  private CompactLineBlock m_labelBlock;
  private CompactLineBlock m_textBlock;

  public CompactLine() {
    setLabelBlock(new CompactLineBlock());
    setTextBlock(new CompactLineBlock());
  }

  public CompactLine(String label, String text) {
    setLabelBlock(new CompactLineBlock(label, null));
    setTextBlock(new CompactLineBlock(text, null));
  }

  public CompactLine(IHeaderCell headerCell, ICell cell) {
    setLabelBlock(convertHeaderCellToBlock(headerCell));
    setTextBlock(convertCellToBlock(cell));
  }

  protected CompactLineBlock convertHeaderCellToBlock(IHeaderCell cell) {
    CompactLineBlock block = new CompactLineBlock();
    if (cell != null) {
      block.setText(cell.getText());
      block.setIcon(cell.getIconId());
      block.setEncodeHtmlEnabled(!cell.isHtmlEnabled());
    }
    return block;
  }

  protected CompactLineBlock convertCellToBlock(ICell cell) {
    CompactLineBlock block = new CompactLineBlock();
    if (cell != null) {
      block.setText(cell.getText());
      block.setIcon(cell.getIconId());
      block.setEncodeHtmlEnabled(!cell.isHtmlEnabled());
    }
    return block;
  }

  public CompactLineBlock getLabelBlock() {
    return m_labelBlock;
  }

  public void setLabelBlock(CompactLineBlock block) {
    Assertions.assertNotNull(block);
    m_labelBlock = block;
  }

  public CompactLineBlock getTextBlock() {
    return m_textBlock;
  }

  public void setTextBlock(CompactLineBlock block) {
    Assertions.assertNotNull(block);
    m_textBlock = block;
  }

  public String build() {
    String label = getLabelBlock().build();
    if (StringUtility.hasText(label)) {
      label += ": ";
    }
    String value = getTextBlock().build();
    if (!StringUtility.hasText(value)) {
      return "";
    }
    return HTML.div(
            span(raw(label)).cssClass("compact-cell-line-label"),
            span(raw(value)).cssClass("compact-cell-line-value"))
        .cssClass("compact-cell-line").toHtml();
  }
}
